File: 20210906-java-fibonacci/src/test/java/com/katas/FibonacciSequenceTest.java
package com.katas;

import org.junit.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class FibonacciSequenceTest {

    private final FibonacciSequence fibonacciSequence = new FibonacciSequence();

    @Test
    public void first_number_in_sequence_is_zero() {
        assertThat(fibonacciSequence.numberAt(0)).isZero();
    }

    @Test
    public void second_number_in_sequence_is_one() {
        assertThat(fibonacciSequence.numberAt(1)).isOne();
    }

    @Test
    public void third_number_in_sequence_is_one() {
        assertThat(fibonacciSequence.numberAt(2)).isOne();
    }

    @Test
    public void fourth_number_in_sequence_is_two() {
        assertThat(fibonacciSequence.numberAt(3)).isEqualTo(2);
    }

    @Test
    public void sixth_number_in_sequence_is_five() {
        assertThat(fibonacciSequence.numberAt(5)).isEqualTo(5);
    }
}

File: 20210321-kotlin-fibonacci/src/test/kotlin/com/katas/FibonacciSequenceTest.kt
package com.katas

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class FibonacciSequenceTest {
    private val fibonacciSequence = FibonacciSequence()

    @Test
    internal fun `first number in sequence is zero`() {
        assertThat(fibonacciSequence.numberAt(0)).isZero()
    }

    @Test
    internal fun `second number in sequence is one`() {
        assertThat(fibonacciSequence.numberAt(1)).isOne()
    }

    @Test
    internal fun `third number in sequence is one`() {
        assertThat(fibonacciSequence.numberAt(2)).isOne()
    }

    @Test
    internal fun `fourth number in sequence is two`() {
        assertThat(fibonacciSequence.numberAt(3)).isEqualTo(2)
    }

    @Test
    internal fun `sixth number in sequence is five`() {
        assertThat(fibonacciSequence.numberAt(5)).isEqualTo(5)
    }
}
